File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers: list, target):
    index = 0
    
    def dfs(index, current):
        if index == len(numbers):
            if current == 0:
                return 1
            else:
                return 0
        
        num = numbers[index]
        index += 1
        
        return dfs(index, (current-num)) + dfs(index, (current+num))
        
    
    return dfs(index, target)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    
    func recursive(index: Int, sum: Int) -> Int {
        if index == numbers.count {
            return target == sum ? 1 : 0
        }
        
        let left = recursive(index: index + 1, sum: sum + numbers[index])
        let right = recursive(index: index + 1, sum: sum - numbers[index])
        
        return left + right
    }
    
    return recursive(index: 0, sum: 0)
}